src/switchboard.rs

```diff
@@ -263,14 +259,14 @@ impl Switchboard {
         })
     }
 
-    pub fn get_users(&self, room: &RoomId) -> HashSet<&UserId> {
-        let mut result = HashSet::new();
-        for session in self.publishers_occupying(room) {
-            if let Some(joined) = session.join_state.get() {
-                result.insert(&joined.user_id);
-            }
-        }
-        result
+    pub fn get_room_users(&self, room: &RoomId) -> impl Iterator<Item = &UserId> {
+        self.publishers_occupying(room).iter().filter_map(|s| {
+            s.join_state.get().map(|j| &j.user_id)
```

Comment:
I was going to say what the behavior here when `s.join_state.get()` return null (create_session and setup_media have been called but not yet process_join), but `publishers_occupying` is using the `publishers_by_room` HashMap and we know that `join_state` is set at this time. So all good.